auth.go

```diff
@@ -62,6 +67,42 @@ func (a *Auth) CheckPermission(scope, userID string) error {
 	return ErrNotAllowed
 }
 
+// GetUsers returns a list of userIDs having one or more permission scopes
+func (a *Auth) GetUsers() ([]string, error) {
+	keys, err := a.store.Keys()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to load permissions")
+	}
+
+	a.logger.Debug("Retrieving all userIDs")
+
+	var userIDs []string
+	for _, key := range keys {
+		userID, err := a.userFromKey(key)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to parse userID from key")
+		}
+		userIDs = append(userIDs, userID)
+	}
+
+	return userIDs, nil
+}
+
+// GetUserPermissions returns the permission scopes for a specific user
+func (a *Auth) GetUserPermissions(userID string) ([]string, error) {
+	key := a.permissionsKey(userID)
+	permissions, err := a.loadPermissions(key)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	a.logger.Debug("Retrieving user permissions",
+		zap.String("user_id", userID),
+	)
```

Comment:
I think this debug message should also be moved to the top of the function :)

Reply:
Makes a lot of sense. Updated in https://github.com/go-joe/joe/pull/28/commits/2ce56fdbf7a39e1f4393c68f9cd688a42a2d70c9